builder/dockerfile/builder_test.go

```diff
@@ -0,0 +1,54 @@
+package dockerfile
+
+import (
+	"strings"
+
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/builder/dockerfile/parser"
+)
+
+func TestBuildProcessLabels(t *testing.T) {
+	dockerfile := "FROM scratch"
+	d := parser.Directive{}
+	parser.SetEscapeToken(parser.DefaultEscapeToken, &d)
+	n, err := parser.Parse(strings.NewReader(dockerfile), &d)
+	if err != nil {
+		t.Fatalf("Error when parsing Dockerfile: %s", err)
+	}
+
+	options := &types.ImageBuildOptions{
+		Labels: map[string]string{
+			"org.a": "cli-a",
```

Comment:
@briceburg The map is always randomized:
```
ubuntu@ubuntu:~/tmp$ cat main.go 
package main

import (
	"fmt"
)

func main() {
	labels := map[string]string{
		"org.a": "cli-a",
		"org.b": "cli-b",
		"org.c": "cli-c",
		"org.d": "cli-d",
		"org.e": "cli-e",
	}
	for k, v := range labels {
		fmt.Println(k, v)
	}
}

ubuntu@ubuntu:~/tmp$ 
ubuntu@ubuntu:~/tmp$ go run main.go 
org.a cli-a
org.b cli-b
org.c cli-c
org.d cli-d
org.e cli-e
ubuntu@ubuntu:~/tmp$ go run main.go 
org.e cli-e
org.a cli-a
org.b cli-b
org.c cli-c
org.d cli-d
ubuntu@ubuntu:~/tmp$ go run main.go 
org.d cli-d
org.e cli-e
org.a cli-a
org.b cli-b
org.c cli-c
ubuntu@ubuntu:~/tmp$ go run main.go 
org.b cli-b
org.c cli-c
org.d cli-d
org.e cli-e
org.a cli-a
ubuntu@ubuntu:~/tmp$
```

I updated the PR to shuffle the order though to make it clear though.